polymath/extensions/pickler.py

```
@@ -90,920 +90,928 @@
+                    if not isinstance(reference[k], numbers.Real):
+                        digit = min(max(_SINGLE_DIGITS, float(digit)), _DOUBLE_DIGITS)
+                elif digit not in {'single', 'double'}:
+                    raise ValueError('invalid pickle digits: ' + repr(digit))
 
-def _validate_pickle_reference(references):
-    """Validate and return the pickle reference values."""
+                new_digits.append(digit)
 
-    original_references = references    # Flake8 thinks this variable is unused # noqa
+        except (ValueError, IndexError, TypeError):
+            raise ValueError('invalid pickle digits: ' + repr(original_digits))
```

Ruff (0.14.7)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)